### PR TITLE
feat: (platform) table - phase 1

### PR DESCRIPTION
**Pull request by olshansky**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3252
#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/table/table.component.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<ng-container *ngIf="rows && rows.length; else emptyTableTemplate">
+    <ng-container *ngIf="tableToolbarComponent">
+        <ng-container *ngTemplateOutlet="tableToolbarComponent.contentTemplateRef; context: { counter: rows.length || 0, size: contentDensity }"></ng-container>
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/table.component.html:3]: in what cases it be work `|| 0`?

File: libs/platform/src/lib/components/table/table.component.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            </tr>
+        </thead>
+        <tbody fd-table-body>
+            <tr fd-table-row
+                *ngFor="let row of rows; let i = index"
+                [attr.aria-selected]="rows[i].checked">
+                <td fd-table-cell
+                    [class.fd-table__cell--checkbox]="selectionMode === 'multiple'"
+                    [class.fd-table__cell--focusable]="selectionMode === 'single'"
+                    [tabIndex]="selectionMode === 'single' ? 0 : -1"
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/table.component.html:39]: plz use enum for 'single' and 'multiple'

File: libs/platform/src/lib/components/table/table.component.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+                    <ng-template #defaultTableCellTemplate>{{ getCellValue(column.key, row) }}</ng-template>
+                </td>
+            </tr>
+        </tbody>
+    </table>
+</ng-container>
+
+<ng-template #emptyTableTemplate>
+    <div>{{ emptyTableMessage }}</div>
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/table.component.html:69]: [innerText]?

File: libs/platform/src/lib/components/table/types/column-align.type.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type ColumnAlign = 'start' | 'end' | 'center';
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/types/column-align.type.ts:1]: why not use enum for it?

File: libs/platform/src/lib/components/table/types/content-density.type.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export type ContentDensity = 'compact' | 'cozy' | 'condensed';
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/types/content-density.type.ts:1]: the same

File: libs/platform/src/lib/components/table/types/selection-mode.type.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export type SelectionMode = 'single' | 'multiple' | 'none';
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/types/selection-mode.type.ts:1]: the same

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/table/table.component.html (hunk 3 above)
Comment (olshansky) [reply libs/platform/src/lib/components/table/table.component.html:69]: `emptyTableMessage ` will come from outside. I'm not sure that developers will provide a pure string _(as I remember `innerText` skip sanitizing and put all from the variable to the block text)_
@Betrozov so, should I put it to `innerText`, what do you think?

Comment (Betrozov) [reply libs/platform/src/lib/components/table/table.component.html]: will add sanitizing or use innerText
choose at your discretion

**Comment by olshansky** [comment]
Merged https://github.com/SAP/fundamental-ngx/pull/3597

**Pull request closed without merging**